src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java

```diff
@@ -577,6 +578,7 @@ public Throwable getLastError() {
 			
 			CompletableFuture<ClientWebSocket> connectTask;
 			synchronized (this.thisLock) {
+				this.closeCalled = true;
```

Comment:
if you wanted this would be a nice place (inside the sync block) to add:

if (this.closeCalled) {
   return CompletableFuture.completed(null);
}